frontend_tests/node_tests/activity.js

```diff
@@ -92,15 +102,47 @@ activity.update_huddles = function () {};
 global.compile_template('user_presence_row');
 global.compile_template('user_presence_rows');
 
+var presence_info = {};
+presence_info[alice.user_id] = { status: 'inactive' };
+presence_info[fred.user_id] = { status: 'active' };
+presence_info[jill.user_id] = { status: 'active' };
+
+presence.presence_info = presence_info;
+
+(function test_get_status() {
+    assert.equal(presence.get_status(page_params.user_id), "active");
+    assert.equal(presence.get_status(alice.user_id), "inactive");
+    assert.equal(presence.get_status(fred.user_id), "active");
+}());
+
+(function test_last_seen_status_from_date() {
+    var last_seen_base_date = new XDate(2016, 2, 29, 10, 30);
+    assert.equal(timerender.last_seen_status_from_date(last_seen_base_date.clone().addSeconds(-20), last_seen_base_date), "Last seen just now");
+    assert.equal(timerender.last_seen_status_from_date(last_seen_base_date.clone().addMinutes(-1), last_seen_base_date), "Last seen a minute ago");
+    assert.equal(timerender.last_seen_status_from_date(last_seen_base_date.clone().addMinutes(-2), last_seen_base_date), "Last seen 2 minutes ago");
+    assert.equal(timerender.last_seen_status_from_date(last_seen_base_date.clone().addMinutes(-45), last_seen_base_date), "Last seen 45 minutes ago");
+    assert.equal(timerender.last_seen_status_from_date(last_seen_base_date.clone().addHours(-1), last_seen_base_date), "Last seen an hour ago");
+    assert.equal(timerender.last_seen_status_from_date(last_seen_base_date.clone().addHours(-2), last_seen_base_date), "Last seen 2 hours ago");
+    assert.equal(timerender.last_seen_status_from_date(last_seen_base_date.clone().addHours(-20), last_seen_base_date), "Last seen yesterday");
+    assert.equal(timerender.last_seen_status_from_date(last_seen_base_date.clone().addDays(-1), last_seen_base_date), "Last seen yesterday");
+
+    var date_2_days_before = last_seen_base_date.clone().addDays(-2);
+    assert.equal(timerender.last_seen_status_from_date(date_2_days_before, last_seen_base_date), "Last seen on " + date_2_days_before.toString("MMM\xa0dd"));
```

Comment:
I would avoid doing calculations for your expected values.  What if your logic for calculating the expected dates is the same broken logic as the actual function?  Instead, just hand compute the dates and put actual dates in the string.